### DISCUSSION TITLE
Panels to record the travel time

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#670] [Q&A] [answered]
New panels to record the travel time will be available in the next Cruiser.

- They work like the [travel distance panels](https://github.com/devemux86/cruiser/discussions/494)

- You can press the panels for their menu:
  - Start / stop the recording
  - Set a custom travel time
  - Reset the recorded time

- Visibility is set in "Settings | Application | Panels"

- During recording their icons are _filled_

For example in the below image, only "travel distance / time 2" panels are recording.

<img src="https://github.com/user-attachments/assets/772c5139-9696-4883-b770-09147669b497" width="50%"/><img src="https://github.com/user-attachments/assets/1ca90426-92b9-4c94-9a99-2107482dcd92" width="50%"/>

**Comment by devemux86** [comment]
The same new functions will also be available on the existing travel distance panels.

**Comment by bios4** [comment]
What should we expect now in terms of "record start/pause/stop" panels?

In the image above only one button is shown (my preference).
[Here](https://github.com/devemux86/cruiser/discussions/660#discussioncomment-11224806) you show two of them (which I don't really like on first sight)...

**Reply by devemux86** [reply]
There are 2 travel distance panels and 2 travel time panels. They record the traveled distance / time.
Not the GPS track, just the value, like the existing 2 travel distance panels.
Although it would be less confusing if we have just 1 pair of distance / time.

GPS recording panels are different, they record the GPS track.
They now split into distance and time and are discussed here:
- https://github.com/devemux86/cruiser/discussions/660

**Reply by kds112** [reply]
Probably we just need a new beta release to test the panels and their behaviour in "real life"...

**Reply by devemux86** [reply]
Soon we will have version **5.1 (Beta)**.
UI must be finalized in order to move forward with the Chinese language.

**Reply by bios4** [reply]
> Probably we just need a new beta release to test the panels and their behaviour in "real life"...

👍 

I definitely lost track now what will be available and which panel will record what (time, distance, GPS track).

**Reply by devemux86** [reply]
- The panels named "GPS recording ..." record the GPS track and display the recorded distance / time.
Like the existing GPS recording panel in **5.0**. Their menu is the main "GPS recording" menu.

- The 2 pairs of panels named "Travel ..." record only the distance / time values and are independent.
Like the existing 2 distance panels in **5.0**. Their menu just has set / reset options.

**Comment by Ciutnik** [comment]
> When recording their icon is filled

Good idea 👍 But maybe it would be more obvious if the icon alternated between empty/filled every second? Would this cause nystagmus and seizures? 😉

**Reply by devemux86** [reply]
It's better to check what's happening on the screen with a quick glance,
without losing information because we're looking at the wrong frame of an animation.

We use numbers in icons because we have 2 panels of traveled distance / time.
Otherwise we could choose better icons from [here](https://fonts.google.com/icons?icon.set=Material+Icons&icon.style=Filled).

So the question again: do we really need 2 panels of traveled distance / time in the same profile?
Profiles store this data, we could switch profile to measure different distance / time.

**Reply by Ciutnik** [reply]
> It's better to check what's happening on the screen with a quick glance

Yes, you are right.

> do we really need 2 panels of traveled distance / time in the same profile?

As for me, I would prefer it this way. I use one counter to check the kilometers traveled since refueling, because I don't have a fuel gauge. (not to mention the rocking of the motorcycle from side to side, which we once recalled here 🙂). The second one may be needed as a daily counter.

**Reply by bios4** [reply]
The two independent trip counters (distance!) are indeed great. Everything else discussed here and requested recently is nothing I personally missed up to now...

**Comment by devemux86** [comment]
- **Cruiser 5.0** records the travel distance in navigation / follow automatically,
when we enable their display in "Settings | Application | Panels".

- ~**Cruiser 5.1 (Beta)** has separate settings for display vs record:~
  - ~Display in "Settings | Application | Panels".~
  - ~Record in navigation / follow in "Settings | Navigation | Panels".~

~Is the split implementation in Cruiser 5.1 (Beta) usable or complicated and prefer to return to the state of Cruiser 5.0?~

**Reply by bios4** [reply]
> complicated and prefer to return to the state of Cruiser 5.0?

Definitely this ^^ from my POV! 😃

**Reply by Ciutnik** [reply]
Does this mean it would allow distance/time to be recorded in the background, regardless of whether the panel is displayed?

**Reply by devemux86** [reply]
@Ciutnik 
If we revert it, it will work like in 5.0: the panels must be set as visible in navigation / follow to record their values.
Navigation can always run in background, with closed screen, etc. if you mean this.

**Reply by devemux86** [reply]
@Ciutnik In 5.0 their visibility in navigation / follow is also their "switch" to record in these modes.

**Reply by Ciutnik** [reply]
Sorry, I wasn't clear. I meant the new functionality:

> Cruiser 5.1 (Beta) has separate settings for display vs record

Will the trip counters (distance and time) not require the panel shown on the screen? That is, they can count in the background?

**Reply by devemux86** [reply]
@Ciutnik In 5.1 (Beta) the visibility and the record switches are separate (see [above](https://github.com/devemux86/cruiser/discussions/670#discussioncomment-11304700)).
So the panels can be hidden and still be set to record in these modes.

**Reply by Ciutnik** [reply]
Thanks. I thought so but wanted to make sure. In image 2 I don't know what is hidden under the "Record distance / time" item. Somehow "Counters 1" and "Counters 2" must be separated?

**Reply by devemux86** [reply]
@Ciutnik See the [1st post](https://github.com/devemux86/cruiser/discussions/670#discussion-7468798) with all screenshots.
The counters are separated in the settings.

**Reply by Ciutnik** [reply]
Oh my! I can check all this in the version I already have installed! 😃 I thought it was the future, but it's the present. I'm a little lost 😁

**Reply by Ciutnik** [reply]
Sorry, one more question: if the counter starts automatically, can it be stopped manually? I only see Edit and Reset.

**Reply by devemux86** [reply]
@Ciutnik Record / stop are currently available when automatic modes are off.
Also in the browse map mode.

We need first to decide what implementation is more simple for most users.
Cruiser 5.0 or 5.1 (Beta) with separate settings, see my question [above](https://github.com/devemux86/cruiser/discussions/670#discussioncomment-11304700).

Then we can improve the current implementation.

**Reply by bios4** [reply]
> We need first to decide what implementation is more simple for most users.

I think this question is already answered with this discussion here... 😉

> I'm a little lost 😁

Me too.

To be honest I have absolutely no idea what is the benefit of having all this functions separated, recording in the background while having it not displayed (or vice versa), values to be reset manually (or automatically), having time and distance recording separately (or not), etc. pp.

**Reply by devemux86** [reply]
We can always revert to the implementation of Cruiser 5.0 with simple common settings in one location.

---

About the new  time panels, I implemented them because I thought they were the suggestion from here:
- https://github.com/devemux86/cruiser/discussions/660

If they are not needed, we can remove them to avoid complicating users with too many options.
And return to the 2 travel distance panels like before.

**Reply by GuG1050** [reply]
Not needed for me, please keep it simple as before.

**Reply by Ciutnik** [reply]
I honestly do not understand the dissatisfaction with the new recording option. After all, it is an option, moreover, disabled by default. So if you do not change anything in the settings nothing will change in use.

I will admit that at first I was lost, but this was due to my lack of knowledge of how to use this option. Now I know and I like it. It is not difficult to understand 🙂

Only I have a comment on the two things I already mentioned: probably should not give up the priority of manual control over automatic. 
The second thing is the naming. The word "Record", in my opinion, fits more with "saving" something (such as a GPS track). Here it is more like "counting". But this is my personal opinion.

**Comment by devemux86** [comment]
Since the travel time panels were already implemented, we can offer them to those who may wish to use them.

Instructions and images are in the first post above:
- https://github.com/devemux86/cruiser/discussions/670#discussion-7468798

**Comment by devemux86** [comment, marked as answer]
- Panels to record the travel time are available in https://github.com/devemux86/cruiser/discussions/820
